File: spo/spo/report/spo_fallbibliothek/spo_fallbibliothek.js
// Copyright (c) 2016, libracore and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["SPO Fallbibliothek"] = {
	"filters": [

	],
    get_datatable_options(options) {
		delete options['cellHeight']
		// change datatable options
		return Object.assign(options, {
			dynamicRowHeight: true
		});
	},
	onload: function(report) {
		report.page.add_inner_button(__("Extend Row"), function() {
			$(".dt-row.vrow").each(function(){
                $(this).addClass("extend-row");
            });
		});
	},
    "formatter": function (value, row, column, data, default_formatter) {
       value = default_formatter(value, row, column, data);
       if (column.id == "anonym") {
            if (value.includes("MEB-")) {
                value = "<span onclick=" + '"goto_med_ber(' + "'" + value + "'" + ');"' + ">" + value + "</span>";
            }
            if (value.includes("TRI-")) {
                value = "<span onclick=" + '"goto_triage(' + "'" + value + "'" + ');"' + ">" + value + "</span>";
            }
       }
       return value;
    }
};

function goto_triage(triage) {
    frappe.call({
        "method": "spo.spo.doctype.triage_anonym.triage_anonym.create_view",
        "args": {
            "triage": triage
        },
        "async": false,
        "callback": function(response) {
            frappe.set_route("Form", "Triage Anonym", response.message);
        }
    });
}

function goto_med_ber(med_ber) {
    frappe.call({
        "method": "spo.spo.doctype.med_ber_anonym.med_ber_anonym.create_view",
        "args": {
            "med_ber": med_ber
        },
        "async": false,
        "callback": function(response) {
            frappe.set_route("Form", "Med Ber Anonym", response.message);
        }
    });
}

File: spo/spo/report/spo_fallbibliothek/spo_fallbibliothek.py
# Copyright (c) 2013, libracore and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _

def execute(filters=None):
    columns, data = [
        {"label": _("Klicken für anonyme Ansicht"), "fieldname": "anonym", "fieldtype": "Data"},
        {"label": _("Problemstellung"), "fieldname": "problemstellung", "fieldtype": "Text Editor"},
        {"label": _("Diagnose"), "fieldname": "diagnose", "fieldtype": "Text Editor"},
        {"label": _("Fragestellung"), "fieldname": "fragestellung", "fieldtype": "Text Editor"},
        {"label": _("Berater*in"), "fieldname": "berater_in", "fieldtype": "Data"}
        ], []
    
    med_bers = frappe.db.sql("""SELECT `name`, `problemstellung`, `diagnose`, `owner` FROM `tabMedizinischer Bericht`""", as_dict=True)
    for med_ber in med_bers:
        _data = []
        _data.append(med_ber.name)
        _data.append(med_ber.problemstellung)
        _data.append(med_ber.diagnose)
        _data.append('---')
        _data.append(med_ber.owner)
        data.append(_data)
    
    triagen = frappe.db.sql("""SELECT `name`, `problemstellung`, `fragestellung_anwalt`, `owner` FROM `tabTriage`""", as_dict=True)
    for triage in triagen:
        _data = []
        _data.append(triage.name)
        _data.append(triage.problemstellung)
        _data.append('---')
        _data.append(triage.fragestellung_anwalt)
        _data.append(med_ber.owner)
        data.append(_data)
    
    return columns, data
